connect/src/routes/tokenBridge/automatic.ts

```diff
@@ -181,10 +181,23 @@ export class AutomaticTokenBridgeRoute<N extends Network>
     if (params.options && params.options.nativeGas > 0) {
       const dtb = await request.toChain.getAutomaticTokenBridge();
       // the maxSwapAmount is in destination chain decimals
-      const maxSwapAmount = await dtb.maxSwapAmount(request.destination.id.address);
+      let maxSwapAmount = await dtb.maxSwapAmount(request.destination.id.address);
+      const redeemableAmountTruncated = amount.truncate(
+        amount.fromBaseUnits(redeemableAmount, amt.decimals),
+        TokenTransfer.MAX_DECIMALS,
+      );
+      const dstDecimals = await request.toChain.getDecimals(request.destination.id.address);
+      const dstAmountReceivable = amount.units(
+        amount.scale(redeemableAmountTruncated, dstDecimals),
+      );
+      if (dstAmountReceivable < maxSwapAmount) {
+        // can't swap more than the receivable amount
+        maxSwapAmount = dstAmountReceivable;
+      }
       const scale = 10000;
```

Comment:
```suggestion
      const GAS_SCALE = 10000;
```

Reply:
I agree that this name is more descriptive and we generally use uppercase for constants (so this suggestion is definitely an improvement), but I'm going to leave it as is for now so not add more changes to the PR that aren't absolutely necessary. Thank you for the feedback though, maybe we can revisit this